src/train/src/learn.rs

```diff
@@ -0,0 +1,46 @@
+use float::Float;
+use hypothesis::hypothesis;
+use indicatif::ProgressIterator;
+use model::Model;
+
+use crate::{prepare::GroupedDatasets, Args};
+
+pub fn learn(arg: &Args, grouped_datasets: &GroupedDatasets, mut model: Model) -> Model {
+	for (label, datasets) in grouped_datasets.iter() {
+		model.weights.insert(
+			label.clone(),
+			vec![0.0; datasets.training.first().expect("empty row").len()],
+		);
+	}
+
+	for _ in (0..arg.iteration).progress() {
+		for (label, datasets) in grouped_datasets {
+			for (model_label, thetas) in &mut model.weights {
+				*thetas = guess(
+					thetas,
+					&datasets.training,
+					arg.learning_rate,
+					label == model_label,
+				);
+			}
+		}
+	}
```

Comment:
update thetas after one iteration, only add them together at end of iteration of at the end of batch